shogun-lib/src/main/java/de/terrestris/shogun/lib/graphql/controller/BaseGraphQLController.java

```diff
@@ -0,0 +1,185 @@
+                "available", id));
+        }
+
+        return this.service.findRevisions(persistedEntity.get());
+    }
+
+    public Optional<Revision<Integer, E>> findRevision(Long id, Integer revId) {
+        Optional<E> persistedEntity = this.service.findOne(id);
+
+        if (persistedEntity.isEmpty()) {
+            throw new EntityNotAvailableException(String.format("Entity with ID %s is not " +
+                "available", id));
+        }
+
+        Optional<Revision<Integer, E>> revision = this.service.findRevision(persistedEntity.get(), revId);
+
+        if (revision.isEmpty()) {
+            throw new EntityNotAvailableException(String.format("Revision %s of entity with ID %s is not " +
+                "available", revId, id));
+        }
+
+        return revision;
+    }
+
+    public List<E> findAllByIds(List<Long> ids) {
+        return this.service.findAllById(ids);
+    }
+
+    public E create(Serializable createEntity) {
+        E entity = this.deserializeInput(createEntity);
+
+        return this.service.create(entity);
+    }
+
+    public E update(Long id, Serializable updateEntity) {
+        Optional<E> persistedEntity = this.service.findOne(id);
+
+        if (persistedEntity.isEmpty()) {
+            throw new EntityNotAvailableException(String.format("Entity with ID %s is not " +
+                "available", id));
+        }
+
+//        updateEntity.put("id", id);
+
+        E entity = this.deserializeInput(updateEntity);
+
+        E updatedEntity = null;
+        try {
+            updatedEntity = this.service.update(id, entity);
+        } catch (IOException e) {
+            log.error("Error while updating entity with ID {}: {}", id, e.getMessage());
+            log.trace("Full stack trace: ", e);
+        }
+
+        return updatedEntity;
+    }
+
+    public Boolean delete(Long id) {
+        Optional<E> persistedEntity = this.service.findOne(id);
+
+        if (persistedEntity.isEmpty()) {
+            throw new EntityNotAvailableException(String.format("Entity with ID %s is not " +
+                "available", id));
+        }
+
+        try {
+            this.service.delete(persistedEntity.get());
+
+            return true;
+        } catch(Exception e) {
+            return false;
+        }
+    }
+
+//    /**
+//     * Returns the simple class name of the {@link BaseEntity} this abstract class
+//     * has been declared with, e.g. 'Application'.
+//     *
+//     * @return The simple class name.
+//     */
+//    public String getGenericSimpleClassName() {
+//        Class<?>[] resolvedTypeArguments = GenericTypeResolver.resolveTypeArguments(getClass(),
+//            BaseGraphQLController.class);
+//
+//        if (resolvedTypeArguments != null && resolvedTypeArguments.length == 2) {
+//            return resolvedTypeArguments[0].getSimpleName();
+//        } else {
+//            return null;
+//        }
+//    }
```

Comment:
Block can be removed.

Reply:
done

Reply:
see https://github.com/terrestris/shogun/pull/595/commits/a90f5e544f8b0f2096e42910863cb7a5a893858f